kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```diff
@@ -180,22 +183,41 @@ public Optional<SslContext> getUpstreamSslContext() {
     private Optional<SslContext> buildDownstreamSslContext() {
         return tls.map(tlsConfiguration -> {
             try {
-                return Optional.of(tlsConfiguration.key()).map(NettyKeyProvider::new).map(NettyKeyProvider::forServer).orElseThrow().build();
+                var sslContextBuilder = Optional.of(tlsConfiguration.key()).map(NettyKeyProvider::new).map(NettyKeyProvider::forServer)
+                        .orElseThrow();
+
+                return configureTrustProvider(tlsConfiguration).apply(sslContextBuilder.clientAuth(toNettyClientAuth(tlsConfiguration)))
+                        .build();
             }
             catch (SSLException e) {
                 throw new UncheckedIOException(e);
             }
         });
     }
 
+    @NonNull
+    private static ClientAuth toNettyClientAuth(Tls tlsConfiguration) {
```

Comment:
I like the fact that it is this module that knows how to map our ClientAuth values into the Netty ones.

Reply:
Yeah that's the core thing I was aiming for but couldn't think how to explain in comments without just writing the code to do it first 😆